test/models/vote_test.rb

```diff
@@ -0,0 +1,38 @@
+require "test_helper"
+
+describe Vote do
+  describe "validation" do
+    it "doesn't allow user to vote twice for 1 work" do
+      #arrange
+      user = users(:incognito)
+      work = works(:metallica)
+      vote = Vote.create(work_id: work.id, user_id: user.id)
+      vote_2 = Vote.create(work_id: work.id, user_id: user.id)
+      #act+assert
+      expect(vote.valid?).must_equal true
+      expect(vote_2.valid?).must_equal false
+      expect(Vote.all.length).must_equal 1
+    end
+  end
+
+  describe "relations" do
```

Comment:
You should also test that a Vote is invalid without either a user_id or work_id.